File: flightincremental/app/service/booking.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Booking } from '../models/booking.model';
@Injectable({
  providedIn: 'root'
})
export class BookingService {
  
  public baseUrl:string = 'https://8080-cbfadfbbdaa322263798edccecfbdeone.premiumproject.examly.io/api/bookings';
  constructor(private http:HttpClient) { }

  getBookings():Observable<Booking[]> {
    return this.http.get<Booking[]>(this.baseUrl);
  }
  createBooking(booking:Booking):Observable<Booking>{
    return this.http.post<Booking>(this.baseUrl,booking);
  }
  updateBooking(booking:Booking):Observable<Booking>{
    return this.http.put<Booking>(`${this.baseUrl}/${booking.bookingId}`,booking);
  }
  getBookingById(id:number):Observable<Booking>{
    return this.http.get<Booking>(`${this.baseUrl}/${id}`);
  }
  deleteBooking(id:number):Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
  getBookingsByUserId(userId:number):Observable<Booking[]>{
    return this.http.get<Booking[]>(`${this.baseUrl}/user/${userId}`);
  }
  updateBookingStatus(bookingId: number, status: boolean):Observable<any> {
    const payload = {status:status};
    return this.http.patch<any>(`${this.baseUrl}/${bookingId}`,payload);
  }
}

File: flightincremental/springapp/service/BookingService.java
package com.examly.springapp.service;

import java.util.List;

import com.examly.springapp.exception.EntityNotFoundException;
import com.examly.springapp.exception.InvalidDateException;
import com.examly.springapp.exception.SeatsExceededException;
import com.examly.springapp.model.Booking;

public interface BookingService {
    Booking createBooking(Booking booking) throws SeatsExceededException,EntityNotFoundException, InvalidDateException;
    Booking getBookingById(Long id) throws EntityNotFoundException;
    List<Booking> getAllBookings();
    Booking updateBooking(Long id, Booking booking) throws EntityNotFoundException;
    List<Booking> getAllBookingsByUserId(int userId);
    Booking updateBookingStatus(long id, boolean status) throws EntityNotFoundException;
}
